reference-implementation/to-upstream-wpts/transform-stream/errors.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+}
+
+const thrownError = new Error('bad things are happening!');
+thrownError.name = 'error1';
+
+promise_test(t => {
+  const results = [];
+
+  const ts = new TransformStream({
+    transform() {
+      throw thrownError;
+    }
+  });
+
+  const reader = ts.readable.getReader();
+
+  results.push(promise_rejects(t, thrownError, reader.read(),
+                               'readable\'s read should reject with the thrown error'));
+
+  results.push(promise_rejects(t, thrownError, reader.closed,
+                               'readable\'s closed should be rejected with the thrown error'));
+
+  const writer = ts.writable.getWriter();
+
+  results.push(promise_rejects(t, thrownError, writer.closed,
+                               'writable\'s closed should be rejected with the thrown error'));
+
+  writer.write('a');
```

Comment:
It's odd that this call to write() is just sitting here with no obvious purpose. I know you didn't add it, and I'm not asking you to remove it. I'm just making a random observation.